### PR TITLE
Add validation rules

### LINKED ISSUE #201: Adding Contact to a Completed or Cancelled Program Shows a Confusing Error Message
**Issue:**

When adding a new Contact to a Program (via a Program Engagement record), results in an error if the Program has a Completed or Cancelled Status. The error reads as "Value does not exist or does not match filter criteria", which is confusing as it doesn't specify the problem lies with the status of the Program. 
Screen Shot 2020-05-01 at 2 29 00 PM

The Program Engagement's "Program" lookup field has a packaged lookup filter that is throwing the error. It's currently configured to have Status = Active or Planned. 

Also, as an Admin I can modify the values in the Program Status field but I'm unable to edit a packaged lookup filter, meaning the filter will no longer match my business criteria. 

**Steps to Repeat:**

1. Create a Program with the status of Completed or Cancelled.
2. Click add the "Add a Contact to Program" button. 
3. Enter Contact, Role, Stage details, and click Save.
Expected: Due to the Status of the Program, error message indicates Contact cannot be added to Program as the Program is Completed or Cancelled. 
Actual: Error message "Value does not exist or does not match filter criteria" is displayed.

**Workaround:**
Until a fix an be put in place, make sure you are adding a Contact to a Planned or Active Program.

### PR DESCRIPTION
**Pull request by bethbrains**
# Critical Changes

# Changes

# Issues Closed
Fixes #201 

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
- [ ] Code does not reference translatable strings in its logic
- [ ] Place holder data is incorporated (Refer to [Case Management Sample Data document](https://quip.com/cbFcAPJF8t0z))
- [ ] Base axe-core JEST test included for any new LWC (No critical violations)
- [ ] CRUD/FLS is enforced in Apex
- [ ] Permission sets are updated to account for CRUD/FLS for new object metadata
- [ ] All modified classes are unit tested (Apex) at 100% coverage
- [ ] UX approval or UX not necessary
- [ ] PR contains draft release notes
- [ ] Attach work item to the pull request with [Lurch](https://salesforce.quip.com/50ZRA5LEWVzH) `**Lurch:attach W-000000` or `**Lurch:add`
- [ ] All **acceptance criteria** have been met
    - [ ] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [ ] QE story level testing completed

### CODE REVIEW

**Comment by bethbrains** [comment]
**lurch: attach W-041172

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000007hZniQAE" target="_blank">W-041172</a>

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HuHPQA0" target="_blank">W-040554</a>

File: force-app/main/default/objects/ProgramEngagement__c/validationRules/PROGRAM_MATCHES_COHORT_PROGRAM.validationRule-meta.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+<ValidationRule xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>PROGRAM_MATCHES_COHORT_PROGRAM</fullName>
+    <active>true</active>
+    <description>The Program must match the Program Cohort&#39;s Program when Program Cohort is populated.</description>
+    <errorConditionFormula>AND(
+!ISBLANK(Program__c),
+!ISBLANK(ProgramCohort__c),
+ProgramCohort__r.Program__c != Program__c
+)</errorConditionFormula>
+    <errorMessage>Select a Program Cohort that matches the Program.</errorMessage>
```
Comment (jjbennett) [inline force-app/main/default/objects/ProgramEngagement__c/validationRules/PROGRAM_MATCHES_COHORT_PROGRAM.validationRule-meta.xml:11]: The Work Item is requesting a filter, assuming we changed to a validation rule but wanted to confirm? Also, our test data is now failing. Should that be a separate work item or handled here?
"Error on record with id 55: FIELD_CUSTOM_VALIDATION_EXCEPTION:Select a Program Cohort that matches the Program.:--"

Comment (bethbrains) [reply force-app/main/default/objects/ProgramEngagement__c/validationRules/PROGRAM_MATCHES_COHORT_PROGRAM.validationRule-meta.xml:11]: Oh shoot that should be fixed here. Let me tackle that this morning. Thanks for the review.

Comment (bethbrains) [reply force-app/main/default/objects/ProgramEngagement__c/validationRules/PROGRAM_MATCHES_COHORT_PROGRAM.validationRule-meta.xml:11]: Done!

**Pull request merged**